include/operations/Operation.hpp

```diff
@@ -97,8 +97,8 @@ namespace qc {
         }
 
         [[nodiscard]] virtual std::vector<dd::Qubit> getUsedQubits() const {
-            auto usedQubits = targets;
-            for (auto control: controls) {
+            auto usedQubits = getTargets();
+            for (auto control: getControls()) {
                 usedQubits.push_back(control.qubit);
             }
             return usedQubits;
```

Comment:
As pointed out above, this would have to change to something like
```c++
const auto& targets = getTargets();
const auto& controls = getControls();
auto usedQubits = std::set{targets.begin(), targets.end()};
for (const auto& control: controls) {
  usedQubits.insert(control.qubit);
}
```

Reply:
Changed the code as suggested